api/include/opentelemetry/logs/logger.h

```diff
@@ -55,65 +54,322 @@ class Logger
-  {
-    // Set severity to the default then call log(Severity, String message) method
-    log(Severity::kDefault, message);
-  }
+  /// Overloaded Log methods, which are simpler than calling a LogRecord directly
 
-  inline void log(Severity severity, nostd::string_view message) noexcept
+  /**
+   * Writes a log.
+   * @param severity The log's severity
+   * @param message The message to log
+   */
+  inline void Log(Severity severity, nostd::string_view message) noexcept
   {
-    // TODO: set default timestamp later (not in API)
-    log(severity, message, core::SystemTimestamp(std::chrono::system_clock::now()));
+    // Create a LogRecord to hold this information
+    LogRecord r;
+    r.severity = severity;
+    r.body     = message;
+
+    // Call the main Log(LogRecord) method
+    Log(r);
   }
 
-  inline void log(Severity severity,
-                  nostd::string_view message,
-                  core::SystemTimestamp time) noexcept
+  /**
+   * Writes a log.
+   * @param severity The log's severity
+   * @param name The name of the log
+   * @param message The message to log
+   */
+  inline void Log(Severity severity, nostd::string_view name, nostd::string_view message) noexcept
   {
-    // creates a LogRecord object with given parameters, then calls log(LogRecord)
+    // Create a LogRecord to hold this information
     LogRecord r;
-    r.severity  = severity;
-    r.body      = message;
-    r.timestamp = time;
+    r.severity = severity;
+    r.name     = name;
+    r.body     = message;
 
-    log(r);
+    // Call the main Log(LogRecord) method
+    Log(r);
   }
 
-  /** Overloaded methods for structured logging**/
-  // TODO: separate this method into separate methods since it is not useful for user to create
-  // empty logs
+  /**
+   * Writes a log.
+   * @param severity The severity of the log, from 1 to 24
+   * @param attributes A key/value object
+   */
   template <class T,
             nostd::enable_if_t<common::detail::is_key_value_iterable<T>::value> * = nullptr>
-  inline void log(Severity severity       = Severity::kDefault,
-                  nostd::string_view name = "",
-                  const T &attributes     = {}) noexcept
+  inline void Log(Severity severity, const T &attributes) noexcept
   {
-    log(severity, name, common::KeyValueIterableView<T>(attributes));
+    // Create a LogRecord to hold this information
+    LogRecord r;
+    r.severity   = severity;
+    r.attributes = nostd::shared_ptr<common::KeyValueIterable>(
+        new common::KeyValueIterableView<T>{attributes});
+
+    // Call the main Log(LogRecord) method
+    Log(r);
   }
 
-  inline void log(Severity severity,
-                  nostd::string_view name,
-                  const common::KeyValueIterable &attributes) noexcept
+  /**
+   * Writes a log.
+   * @param severity The severity of the log, from 1 to 24
+   * @param name The name of the log
+   * @param attributes A key/value object
+   */
+  template <class T,
```

Comment:
nit: for all template class, prefer `template <typename T,`.

Reply:
For this we were following what was done in tracer.h and its [StartSpan template classes](https://github.com/open-telemetry/opentelemetry-cpp/blob/master/api/include/opentelemetry/trace/tracer.h#L46).